onyo/commands/edit.py

```diff
@@ -57,16 +135,26 @@ def edit(args, onyo_root):
         sys.exit(1)
 
     # check and set paths
-    list_of_sources = prepare_arguments(args.asset, onyo_root)
-    # iterate over file list, edit them, add changes
-    for source in list_of_sources:
-        git_filepath = os.path.relpath(source, onyo_root)
-        # change file
-        edit_file(source, onyo_root)
-        # add any changes
-        repo._git(['add', git_filepath])
+    assets = sanitize_assets(args.asset, repo)
+    editor = get_editor(repo.root)
+
+    for asset in assets:
+        if edit_asset(editor, asset):
+            repo.add(asset)
+        else:
+            """
+            If user wants to discard changes, restore the assets state
+            """
+            repo._git(['restore', str(asset)])
+            print(f'"{asset}" not updated.')
 
     # commit changes
     files_staged = repo.files_staged
     if files_staged:
-        repo._git(['commit', '-m', 'edit asset(s).\n\n' + '\n'.join(files_staged)])
+        print(diff(repo))
+        if request_user_response("Save changes? No discards all changes. (y/n) "):
+            repo.commit('edit asset(s).', files_staged)
+        else:
+            for asset in files_staged:
```

Comment:
`git restore` accepts multiple arguments. Rather than a for-loop, you can simply pass them all directly to `git`. That way it's only one `git` call rather than n-edited-assets.